gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/ReporterServiceProxyImpl.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.management.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.gravitee.am.identityprovider.api.User;
+import io.gravitee.am.management.service.AbstractSensitiveProxy;
+import io.gravitee.am.management.service.ReporterPluginService;
+import io.gravitee.am.management.service.ReporterServiceProxy;
+import io.gravitee.am.management.service.exception.ReporterPluginSchemaNotFoundException;
+import io.gravitee.am.model.Reporter;
+import io.gravitee.am.service.exception.ReporterNotFoundException;
+import io.gravitee.am.service.model.NewReporter;
+import io.gravitee.am.service.model.UpdateReporter;
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Titouan COMPIEGNE (titouan.compiegne at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Component
+public class ReporterServiceProxyImpl extends AbstractSensitiveProxy implements ReporterServiceProxy {
+
+    @Autowired
+    private ReporterPluginService reporterPluginService;
+
+    @Autowired
+    private io.gravitee.am.service.ReporterService reporterService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    public Flowable<Reporter> findAll() {
+        return reporterService.findAll().flatMapSingle(this::filterSensitiveData);
+    }
+
+    @Override
+    public Flowable<Reporter> findByDomain(String domain) {
+        return reporterService.findByDomain(domain).flatMapSingle(this::filterSensitiveData);
+    }
+
+    @Override
+    public Maybe<Reporter> findById(String id) {
+        return reporterService.findById(id).flatMap(reporter -> filterSensitiveData(reporter).toMaybe());
```

Comment:
As `reporterService.findById(id)` returns a Maybe, it can be empty. In that case flatMapSingle will throw an error.
We should do the same tihng as https://github.com/gravitee-io/gravitee-access-management/pull/1507/files#diff-2eed3cdefb0682c04dfda0b09f2eca8ff3fb8b64a1b56b7d8e8ed14cf6a20e1aR84

Reply:
Done